### ISSUE TITLE
🐛  By default, Autofixture is not compatible with 'Verify' package

### ISSUE THREAD

**Issue by rui-santos-paysimplex** [AutoFixture/AutoFixture#1485]
### Describe the Bug
I've recently started using [Verify](https://github.com/VerifyTests/Verify) nuget package and I bumped into a problem right away.
For string properties, if you don't specify any value, AutoFixture generates the value with the following format: `{PropertyName}{Guid.NewGuid()}`.

Verify's assertions are based on comparing a previously known value to a new one. If they match, the assertion succeeds, if they don't match, the assertion fails.
Guid values could be a problem for this, because since you always get a new/different one, the previously known value and the new value will always be different, so they will never match, therefore the test will always fail.
Verify has a solution for this, it recognizes the Guid values as Guids and replaces them with a fixed value.
The problem is that, since AutoFixture generates the value with the string (property name) and Guid (for randomness) parts 'stuck' together, Verify can't recognize the guid as a guid. This prevents Verify from being able to scrub out the Guid value.

### Scenario
```
class SomeClass
{
  public string SomeProperty { get; init; }
}

[Fact]
public async Task SomeTestMethod()
{
  var fixture = new Fixture();
  var someClassObject = fixture.Create<SomeClass>();

  await Verifier.Verify(someClassObject);
}
```

### Expected Behavior
The first run will always fail, because it first has to generate the file that will be later used for comparison (it's just how Verify works).
The second run should succeed because the object should be the same as the previous run.
What actually happens is that it fails because all string properties will have new `Guids` in their values. This is because Verify can't recognize them as `Guid` and just assumes that they are 'normal' characters.
I expect the test to succeed because Verify should be able (and it does in other scenarios) to recognize Guids as actual Guids and not just a normal string value.

The solution seems simple to me (because I applied it, and it worked right away).
Instead of auto generating the string properties with the format `{PropertyName}{Guid.NewGuid()}`, generate them with this format, `{PropertyName}_{Guid.NewGuid()}.

The `_` (underscore) separating the PropertyName and the Guid value is enough to allow 'Verify' to recognize the Guid as Guid and be able to scrub it.

### Screenshots
The problem:
![image](https://github.com/user-attachments/assets/ec4e2115-86d8-4640-a16f-a8cbb284d66e)

The solution:
![image](https://github.com/user-attachments/assets/6fd283b6-589e-42c2-87a6-800dfa7e5a79)
![image](https://github.com/user-attachments/assets/a5573489-ab6e-4c9c-8f3c-153896fe03c4)

👍 
![image](https://github.com/user-attachments/assets/6b352d13-53db-4c21-be68-eb3c3017329c)

Thanks!

**Comment by aivascu** [comment]
@Kralizek I need a second opinion on this. What do you think of this change?
I see no harm in making this change, and I think it might actually be useful to distinguish between the seed and the auto-generated string.

**Comment by Kralizek** [comment]
Given the nature and purpose of the library, changing how random strings are generated shouldn't be a breaking change since the shape of the generated strings is not part of the API.

That being said, there might be some people who rely on the current behavior.

If possible, I'd introduce a new string generator that people can switch on using a customization masked as sugar method.

Introducing this in V5 and making it the default behavior in V6 is the safest route.

I can't say whether is necessary to be so cautious, tho.

**Comment by aivascu** [comment]
Yes I have the same reservations. What if someone depends on the current behavior and changing it will throw off their regex?

Also the seeded string generation and member request relays are part of the engine. A sugar method is not possible as the engine parts are immutable after the fixture instance is created.

Even if the default behavior changed, what if the frozen `string` is empty then the generated values would be `PropertyName_` where as previously they'd be `PropertyName`. Do we also add custom logic, to add the separator only when the seed is not empty?

In a situation like this I'd expect the users to create their own fixture type with the engine parts configured appropriately. Then again the customization of the engine parts is not the most trivial thing in AutoFixture, and the `DefaultRelays` constructor overload is inconvenient at best.

I'm now leaning more toward leaving the behavior as is, but adding a constructor overload that would simplify the customization of the engine parts.

Here's the shortest example I could come up with to add the custom parts for the Verify requirements.

```cs
[Fact]
public void UsesCustomEngineParts()
{
    // Arrange
    var fixture = new MyFixture();
    
    // Act
    var result1 = fixture.Create<string>("hello_9023");
    var result2 = fixture.Create<PropertyHolder<string>>();
    
    // Assert
    Assert.StartsWith("hello_9023_", result1);
    Assert.StartsWith("Property_", result2.Property);
}

public class MyFixture : Fixture
{
    public MyFixture()
        : base(new CompositeSpecimenBuilder(
            new CompositeSpecimenBuilder(
                    new VerifyStringSeedRelay()
                    /* other custom parts */)
                .Concat(new DefaultPrimitiveBuilders())
                .Concat(new DefaultRelays())),
            new MultipleRelay())
    {
    }
}

public class VerifyStringSeedRelay : ISpecimenBuilder
{
    public object Create(object request, ISpecimenContext context)
    {
        if (context is null) throw new ArgumentNullException(nameof(context));

        if (!(request is SeededRequest seededRequest
              && seededRequest.Request.Equals(typeof(string))
              && seededRequest.Seed is string seed))
        {
            return new NoSpecimen();
        }

        var containerResult = context.Resolve(typeof(string));
        return containerResult is NoSpecimen
            ? containerResult
            : string.Concat(seed, '_', containerResult);
    }
}
```
